src/main/java/org/ehrbase/fhirbridge/ehr/converter/TherapyCompositionConverter.java

```diff
@@ -0,0 +1,264 @@
+
+
+        if (procedure == null) {
+            return null;
+        }
+
+        GECCOProzedurComposition result = new GECCOProzedurComposition();
+
+        // set feeder audit
+        FeederAudit fa = CommonData.constructFeederAudit(procedure);
+        result.setFeederAudit(fa);
+
+        // Map Kategorie
+
+        result.setKategorie(new ArrayList<>());
+
+        for (Coding coding : procedure.getCategory().getCoding()) {
+            if (coding.getSystem().equals(SNOMED_SYSTEM) && kategorieMap.containsKey(coding.getCode())) {
+                GeccoProzedurKategorieElement element = new GeccoProzedurKategorieElement();
+                element.setValue(kategorieMap.get(coding.getCode()));
+
+                result.getKategorie().add(element);
+            }
+        }
+
+
+        // Map Start Time
+        try {
+
+            if (!procedure.getExtension().isEmpty() && procedure.getExtension().get(0).getValue() instanceof DateTimeType) {
+                result.setStartTimeValue(((DateTimeType) procedure.getExtension().get(0).getValue()).getValueAsCalendar().toZonedDateTime());
+            } else {
+                result.setStartTimeValue(procedure.getPerformedDateTimeType().getValueAsCalendar().toZonedDateTime());
+            }
+
+        } catch (NullPointerException | FHIRException ex) {
+            result.setStartTimeValue(ZonedDateTime.now());
+        }
+
+        switch (procedure.getStatus()) {
+            case UNKNOWN:
+                mapUnknown(procedure, result);
+                break;
+            case NOTDONE:
+                mapNotDone(procedure, result);
+                break;
+            case ENTEREDINERROR:
+                throw new CompositionConversionException("Invalid status");
+            default:
+                mapDone(procedure, result);
+        }
+
+        // ======================================================================================
+        // Required fields by API
+
+        result.setLanguage(Language.DE);
+        result.setLocation("test");
+        result.setSettingDefiningCode(Setting.SECONDARY_MEDICAL_CARE);
+        result.setTerritory(Territory.DE);
+        result.setCategoryDefiningCode(Category.EVENT);
+        result.setComposer(new PartySelf());
+
+        return result;
+    }
+
+    private NameDerProzedurDefiningCode mapNameDerProzedur(Procedure procedure) throws UnprocessableEntityException {
+        Coding coding = procedure.getCode().getCoding().get(0);
+
+        if (coding.getSystem().equals(SNOMED_SYSTEM) && nameDerProzedurMap.containsKey(coding.getCode())) {
+            return nameDerProzedurMap.get(coding.getCode());
+        } else {
+            throw new UnprocessableEntityException("Invalid name of procedure");
+        }
+    }
+
+    private void mapDone(Procedure procedure, GECCOProzedurComposition composition) {
+
+        ProzedurAction durchgefuehrteProzedur = new ProzedurAction();
+
+        try {
```

Comment:
try block is also way too long, i would suggest maybe to return the exception and catch it in the method which calls it

Reply:
Fixed